### ISSUE TITLE
Incorrect instance of ImplicitReturnStatement

### ISSUE THREAD

**Issue by lowrider59** [CodeNarc/CodeNarc#725]
When we have a method like:

```
public someType someMethod() {
  synchronized (someObject) {
    doSomething();
    return something;
  }
}

```
we get a msg like:

"The method someMethod in class SomeClass is missing an explicit return  ImplicitReturnStatement"

which I consider incorrect in this specific construct.

I reported this issue on npm-groovy-lint, but the response I got was, "this is a CodeNarc issue".
See:  
  [https://github.com/nvuillam/npm-groovy-lint/issues/264](https://github.com/nvuillam/npm-groovy-lint/issues/264)

**Comment by chrismair** [comment]
Yes, that does look like a CodeNarc issue.

**Comment by chrismair** [comment]
Fix will be available in CodeNarc v3.3.0
